packages/x-date-pickers/src/PickersTextField/PickersOutlinedInput/PickersOutlinedInput.tsx

```diff
@@ -57,7 +57,7 @@ const PickersOutlinedInputRoot = styled(PickersInputBaseRoot, {
     },
     variants: Object.keys((theme.vars ?? theme).palette)
       // @ts-ignore
-      .filter((key) => (theme.vars ?? theme).palette[key].main)
+      .filter((key) => (theme.vars ?? theme).palette[key]?.main ?? false)
```

Comment:
The `key` can now be `'unknown'` – see https://github.com/mui/material-ui/pull/42147/files#diff-41fa27eaac444c45bc6ea739297c3fd1ae31d5edae90310e731c2df0e38cb7c1R998

cc @mui/explore